### DISCUSSION TITLE
hide passwords sensitive data trino config files

### DISCUSSION THREAD

**Discussion by manjot-aptus** [trinodb/trino#12689] [Q&A]
I am trying to hide the credentials of Catalog files. But not able to do it. If possible help in this matter.

**Comment by ebyhr** [comment]
Did you try environment variables https://trino.io/docs/current/security/secrets.html?

**Reply by manjot-aptus** [reply]
I am not using Trino RPM. But I wanted to use the environmental variable concept to store the user name and password. I tried to export these variables in .bashrc file. But I am not able to work with it. Please help me with some steps.

**Reply by hashhar** [reply]
bashrc is only used for interactive shells. You'll want to set the environment variables in whatever you are using to start the Trino process (systemd, the init script or something else).

**Reply by manjot-aptus** [reply]
I am using Ubuntu. If I am trying to install trino rpm I am getting an issue. I am using alien package to convert rpm to deb or I tried to install it directly using alien but failed in both cases.
![Screenshot from 2022-06-07 14-13-13](https://user-images.githubusercontent.com/85937124/172337022-e52dabb4-ca31-472d-8651-b5d21e5a0100.png)

**Reply by hashhar** [reply]
You can use the tarball directly. No need for the RPM.

Extract the tarball and follow steps at https://trino.io/docs/current/installation/deployment.html

**Reply by manjot-aptus** [reply]
But for environment variables, we need to use the trino file where we will define to preserver environment variables. I am doing it the first time so don't know the directory structure that I need to follow. If  you have any document help me please,

**Reply by hashhar** [reply]
The place where environment variables are defined depends on how you start Trino. e.g. If you the launcher as described in https://trino.io/docs/current/installation/deployment.html then you can do like

```
MY_SECRET_ENV_VAR=my-secret-value bin/launcher start
```

which will make the MY_SECRET_ENV_VAR available only to the Trino process. You can also set the variables in something like `/etc/profile` but that means that the environment variable will be available to all processes running on system.
The way env var is set depends on how you deploy so it varies from system to system (e.g. systemd makes it easier, docker and k8s have their own mechanisms etc.)

**Reply by manjot-aptus** [reply]
Thanks, I will try this solution

**Comment by jerryleooo** [comment]
For JDBC connectors, they support extra credentials in which you can pass the username/password at the runtime instead of writing it in config files, I am working on a document about it: https://github.com/trinodb/trino/pull/11791
